File: helfomat-web/src/main/java/de/helfenkannjeder/helfomat/web/controller/AdminController.java
package de.helfenkannjeder.helfomat.web.controller;

import de.helfenkannjeder.helfomat.infrastructure.batch.ImportJobRunnerService;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * @author Valentin Zickner
 */
@RestController
@RequestMapping("/admin")
public class AdminController {

    private final ImportJobRunnerService importJobRunnerService;

    public AdminController(ImportJobRunnerService importJobRunnerService) {
        this.importJobRunnerService = importJobRunnerService;
    }

    @PostMapping("/import")
    public void startImport() {
        this.importJobRunnerService.run();
    }

}

File: helfomat-web/src/main/kotlin/de/helfenkannjeder/helfomat/web/controller/AdminController.kt
package de.helfenkannjeder.helfomat.web.controller

import de.helfenkannjeder.helfomat.api.approval.AdminApplicationService
import org.springframework.http.MediaType
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

/**
 * @author Valentin Zickner
 */
@RestController
@RequestMapping(path = ["/api"], produces = [MediaType.APPLICATION_JSON_VALUE])
class AdminController(
    private val adminApplicationService: AdminApplicationService
) {

    @GetMapping("/reindex-all-organizations")
    fun reindexAllOrganizations() = adminApplicationService.reloadAllOrganizations()

}